pkg/api/norman/customization/aks/listers.go

```diff
@@ -120,6 +122,94 @@ func (s sortableVersion) Less(a, b int) bool {
 	return version.Compare(s[a], s[b], "<")
 }
 
+type KubernetesUpgradeVersion struct {
+	Version string `json:"version"`
+	Enabled bool   `json:"enabled"`
+}
+
+type KubernetesUpgradeVersions []*KubernetesUpgradeVersion
+
+func (s KubernetesUpgradeVersions) Len() int {
+	return len(s)
+}
+
+func (s KubernetesUpgradeVersions) Swap(a, b int) {
+	s[a], s[b] = s[b], s[a]
+}
+
+func (s KubernetesUpgradeVersions) Less(a, b int) bool {
+	return version.Compare(s[a].Version, s[b].Version, "<")
+}
+
+// listKubernetesUpgradeVersions lists all kubernetes versions listed by AKS Container Service and marks which ones the
+//given cluster can be upgraded to.  A version's `Enabled` flag is true if the cluster can be upgraded to the version
+//in its current state.
+func listKubernetesUpgradeVersions(ctx context.Context, clusterLister v3.ClusterLister, cap *Capabilities) ([]byte, int, error) {
+	// load the target cluster, if the cluster is not found we cannot proceed
+	cluster, err := clusterLister.Get("", cap.ClusterId)
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to load cluster: %v", err)
+	}
+
+	// get the client for aks container service
+	clientContainer, err := NewContainerServiceClient(cap)
+	if err != nil {
+		return nil, http.StatusInternalServerError, err
+	}
+
+	// request a list of orchestrators
+	orchestrators, err := clientContainer.ListOrchestrators(ctx, cap.ResourceLocation, "managedClusters")
+	if err != nil {
+		return nil, http.StatusBadRequest, fmt.Errorf("failed to get orchestrators: %v", err)
```

Comment:
Wouldn't 500 make more sense here? There was some error listing orchestrators.

Reply:
I copied this from other parts of the codebase for these endpoints.  I agree a 500 makes more sense but then we are deviating from the behavior of the other endpoints.  I think we should make a ticket to cleanup the code for these endpoints and make the response consistent for all of them.

Reply:
Ah, I see now. In this case we should favor consistency, so no change is necessary here. For the same reason, I can get behind closing the other 2 comments (including clusterId handling).